library/std/src/os/hurd/raw.rs

```diff
@@ -0,0 +1,81 @@
+//! Hurd-specific raw type definitions
+
+#![stable(feature = "raw_ext", since = "1.1.0")]
+#![deprecated(
```

Comment:
All other unix OS ports have these pub types, because it's used in `library/std/src/os/unix/raw.rs`. Without these pub types, we get:

```
error[E0432]: unresolved import `super::platform::raw::pthread_t`
  --> library/std/src/os/unix/raw.rs:27:9
   |
27 | pub use super::platform::raw::pthread_t;
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `pthread_t` in `os::hurd::raw`
   |
help: consider importing this type alias instead
   |
27 | pub use libc::pthread_t;
   |         ~~~~~~~~~~~~~~~

error[E0432]: unresolved imports `super::platform::raw::blkcnt_t`, `super::platform::raw::time_t`
  --> library/std/src/os/unix/raw.rs:30:32
   |
30 | pub use super::platform::raw::{blkcnt_t, time_t};
   |                                ^^^^^^^^  ^^^^^^ no `time_t` in `os::hurd::raw`
   |                                |
   |                                no `blkcnt_t` in `os::hurd::raw`
   |
   = help: consider importing this type alias instead:
           libc::blkcnt_t
   = help: consider importing this type alias instead:
           libc::time_t

error[E0432]: unresolved imports `super::platform::raw::blksize_t`, `super::platform::raw::dev_t`, `super::platform::raw::ino_t`, `super::platform::raw::mode_t`, `super::platform::raw::nlink_t`, `super::platform::raw::off_t`
  --> library/std/src/os/unix/raw.rs:33:32
   |
33 | ...m::raw::{blksize_t, dev_t, ino_t, mode_t, nlink_t, off_t};
   |             ^^^^^^^^^  ^^^^^  ^^^^^  ^^^^^^  ^^^^^^^  ^^^^^ no `off_t` in `os::hurd::raw`
   |             |          |      |      |       |
   |             |          |      |      |       no `nlink_t` in `os::hurd::raw`
   |             |          |      |      no `mode_t` in `os::hurd::raw`
   |             |          |      no `ino_t` in `os::hurd::raw`
   |             |          no `dev_t` in `os::hurd::raw`
   |             no `blksize_t` in `os::hurd::raw`
   |
   = help: consider importing this type alias instead:
           libc::blksize_t
   = help: consider importing this type alias instead:
           libc::dev_t
   = help: consider importing this type alias instead:
           libc::ino_t
   = help: consider importing this type alias instead:
           libc::mode_t
   = help: consider importing this type alias instead:
           libc::nlink_t
   = help: consider importing this type alias instead:
           libc::off_t
```

Sure, the unix piece should be cleaned up towards not using them, but I don't think it's the role of an OS-adding commit to do this? (but rather a commit that does so and along the way drops all the deprecated pubs from all OSes)

I however noticed that the stat pub types are not used any more, so I dropped them indeed.